cachi2/core/package_managers/pip.py

```diff
@@ -87,11 +89,16 @@ def fetch_pip_source(request: Request) -> RequestOutput:
             package.requirements_files,
             package.requirements_build_files,
         )
-
-        components.append(Component.from_package_dict(info["package"]))
+        purl = _generate_purl(info["package"], request.source_dir.join_within_root(package.path))
```

Comment:
Can we do the `join_within_root` only once (before calling resolve_pip)? It's not the cheapest operation